### PR TITLE
feat(FEC-13116): set text contrast color according to base color

### PR DESCRIPTION
**Pull request by semarche-kaltura**
### Description of the Changes

solves: https://kaltura.atlassian.net/browse/FEC-13116

Screenshot 2023-06-16 at 17 34 56
Screenshot 2023-06-16 at 17 37 24

### CODE REVIEW

File: src/utils/themes-manager.js [hunk 1]
```diff
@@ -61,7 +91,9 @@ export class ThemesManager {
 
     for (const color in config.colors) {
       if (ACCENT_AND_ACKNOWLEDGEMENT_COLORS.includes(color)) {
-        this.setAccentOrAcknowledgementColor(color, config.colors[color]);
+        const colorValue = config.colors[color];
+        this.setAccentOrAcknowledgementColor(`${color}-${textContrastSuffix}`, this.makeTextContrastColor(colorValue));
+        this.setAccentOrAcknowledgementColor(color, colorValue);
```
Comment (SivanA-Kaltura) [inline src/utils/themes-manager.js:96]: why is this called twice ?

File: src/utils/themes-manager.js [hunk 2]
```diff
@@ -49,6 +51,34 @@ export class ThemesManager {
     }
   }
 
+  /**
+   * Makes color value for text.
+   * @param {string} inputColor  - css color in HEX format
+   * @returns {string} hex  - css color in HEX format
+   */
+  makeTextContrastColor(inputColor: string) {
+    const hsl = hexToHsl(inputColor);
```
Comment (SivanA-Kaltura) [inline src/utils/themes-manager.js:60]: what is this calculation based on ?

File: src/utils/themes-manager.js [hunk 3]
```diff
@@ -26,6 +26,8 @@ const dynamicColoredIconsSvgUrlVars = [
   '--playkit-icon-quality-8K-active-url'
 ];
 
+const textContrastSuffix = 'text-contrast';
```
Comment (SivanA-Kaltura) [inline src/utils/themes-manager.js:29]: better use caps if it's a constant

File: src/utils/color-format-convertors.js [hunk 4]
```diff
@@ -81,3 +81,59 @@ export function rgbToHsl(rgb) {
...
+
+/**
+ * Convert css color form HSL to HEX format.
+ * @param {number} h  - hue
+ * @param {number} s  - saturation
+ * @param {number} l  - lightness
+ * @returns {string} hex  - css color in HEX format
+ */
+export function hslToHex(h, s, l) {
+  s /= 100;
```
Comment (SivanA-Kaltura) [inline src/utils/color-format-convertors.js:93]: where is this copied from ?

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]

File: src/utils/themes-manager.js (hunk 1 above)
Comment (semarche-kaltura) [reply src/utils/themes-manager.js:96]: targets are different
1st call for text-color
2nd call for brand color

File: src/utils/themes-manager.js (hunk 2 above)
Comment (semarche-kaltura) [reply src/utils/themes-manager.js:60]: calculation taken from https://codepen.io/hillelkaltura/pen/NWOXMWM?editors=0110
Dana approved such calculation as part of Studio and KVE projects

File: src/utils/color-format-convertors.js (hunk 4 above)
Comment (semarche-kaltura) [reply src/utils/color-format-convertors.js:93]: https://codepen.io/hillelkaltura/pen/NWOXMWM?editors=0110

File: src/utils/themes-manager.js (hunk 3 above)
Comment (semarche-kaltura) [reply src/utils/themes-manager.js]: addressed

**Review by SivanA-Kaltura — APPROVED** [review]

**Review by lianbenjamin — APPROVED** [review]

**Pull request merged**